controllers/argocd/applicationset.go

```diff
@@ -176,6 +180,12 @@ func (r *ReconcileArgoCD) reconcileApplicationSetDeployment(cr *argoprojv1a1.Arg
 	}}
 
 	if existing := newDeploymentWithSuffix("applicationset-controller", "controller", cr); argoutil.IsObjectFound(r.Client, cr.Namespace, existing.Name, existing) {
+		if applicationSetDisabled {
+			if err := r.Client.Delete(context.TODO(), existing); err != nil {
```

Comment:
can we have a `log.Info` here that mentions that the applicationSet deployment is being deleted because applicationSet was disabled?